src/main/java/teammates/common/util/Config.java

```diff
@@ -95,7 +95,11 @@ private String getGcsBucketname() {
     }
 
     private String getAppUrl() {
-        return props.getProperty("app.url");
+        return Url.trimTrailingSlash(props.getProperty("app.url"));
+    }
+    
+    public static AppUrl getAppUrl(String url) {
+        return new AppUrl(APP_URL, url);
     }
```

Comment:
You are right, it is not a pure factory method, because Config is not a factory class. Over here, we should think from the point of view of the Config class. Its job is to represent config values, not construct URLs. It doesn't sound right when is receives parameter to process. So the usage will be like this: 
```java
AppUrl url = Config.getAppUrl().setRelativePath(relativePath);
url = Config.getAppUrl().withStudentEmail(studentName).withUser(userId);
url = new AppUrl(fullUrl); //no involvement of Config because the caller seems to know the app url already
```